Jetbot_ASR_client:
  ros__parameters:
    ASR_topic: "/jetbot_voice/transcripts"
    TTS_topic: "/jetbot_TTS/transcripts"
    start: true
    command_nodes: ["/Jetbot_tool_voice_copilot"]
    # jetbot_commands: syntax: ['ASR keyword', 'command nodes index:command keyword']
    # jetbot_commands: example: ['away','0:self-driving']
    jetbot_commands: >
      [
        ['silence','0:silence'], ['start','0:start'], ['stop','0:stop'],['stuff','0:stop'],
        ['left','0:left'], ['right','0:right'], ['ahead', '0:forward'], ['behind', '0:backward'],
        ['away','0:self-driving'], ['back','0:follow'], ['come','0:follow']
      ]
    jetbot_chat: >
      [
        ['hello', 'This is jetbot. How can I help you'],
        ['name', 'This is jetbot. How can I help you'],
        ['bye', 'Take care, talk to you later']
      ]
Jetbot_TTS:
  ros__parameters:
    TTS_topic: "/jetbot_TTS/transcripts"
    start: true
Jetbot_tool_voice_copilot:
  ros__parameters:
    ASR_topic: "/jetbot_voice/transcripts"
    TTS_topic: "/jetbot_TTS/transcripts"
    start: true
    command_nodes: ["/laser_avoidance","/detect_copilot", "/follow_copilot"]
    laser_topic: "/jetbot1/scan"
    cmd_vel_topic: "/jetbot1/cmd_vel"
    global_ns: true
    base_frame: "jetbot1/base_footprint"
    # jetbot_commands: syntax: ['command keyword', 'command name : distance/angle']
    # command_keyword - Jetbot_ASR_client:jetbot_commands:[index:command_keyword]
    # jetbot_commands: example: ['forward','move:0.5']
    jetbot_commands: >
      [
        ['forward','move:0.5'], ['backward','move:-0.5'],
        ['left','turn:90'], ['right','turn:-90'],
        ['stop', 'stop:0']
      ]
    # jetbot_tasks: syntax: [task keyword, 'command_nodes index, parameter name']
    # jetbot_tasks: ["follow", "2:follow_detect"]
    jetbot_tasks: >
      [
        ['follow', '2:follow_detect'], ['self-driving', '0:start']
      ]
    angle: 30
    angular_tolerance: 3.0
    linear_tolerance: 0.05
    #  Jetson DNN inference ROS2 detectnet node
    node_name: "/detectnet/detectnet"
    class_labels: "class_labels_3185375291238328062"
    overlay_topic: "/detectnet/overlay"
    detect_topic: "/detectnet/detections"
    fov: 90.0
    score: 0.5
    tracking_objects: ['person']
    detect_timeout: 12
